### ISSUE TITLE
Template functions don't work when writing a new message in a newsgroup account

### ISSUE THREAD

**Issue by axxisd** [RealRaven2000/SmartTemplate4#311]
When writing a new message to a specific newsgroup, the new message template is not used.

Currently, you must use CTRL-N (or select dropdown "New > Message", or button "+ New Message") twice in order for the template to show up, but then at that point you are not writing to a specific newsgroup any more (the "Newsgroup" field goes blank).

If it's a completely different scenario than what it was originally meant for, then possibly adding new 'Account Template' fields for "Post" and "Followup" templates?

* At the moment, the "Reply" template works perfectly fine when using "Followup."  So I had just assumed "Write" would work for any new message, also.

**Comment by RealRaven2000** [comment]
> When writing a new message to a specific newsgroup, the new message template is not used.


I guess you are meaning Account templates ? External templates should always work:

![image](https://github.com/user-attachments/assets/77d332ac-3a6b-486f-bc36-4ee85a000ff4)

- just took me a while to find and sub to a newsgroup again. apparently everything is broken when we write a new mail to the newsgroup, even the "change template " and "smart snippets" dropdowns are not populated. This needs some investigation...

I believe this should have worked at one stage. strangely  no errors are shown in error console either... well not quite, if we click on "New Message" I get this output:

```
window.controllers/Controllers is deprecated. Do not use it for UA detection. 2 blank
Please use ChromeUtils.defineLazyGetter instead of XPCOMUtils.defineLazyGetter. XPCOMUtils.defineLazyGetter will be removed soon. 4 XPCOMUtils.sys.mjs:47:13
SmartTemplates st-composer.js - onLoad() smartTemplate-util.js:791:12
Please use ChromeUtils.defineLazyGetter instead of XPCOMUtils.defineLazyGetter. XPCOMUtils.defineLazyGetter will be removed soon. 4 XPCOMUtils.sys.mjs:47:13
SmartTemplates smartTemplate-composer.js smartTemplate-util.js:791:12
NS_ERROR_FAILURE: Component returned failure code: 0x80004005 (NS_ERROR_FAILURE) [nsIMessenger.msgHdrFromURI] smartTemplate-overlay.js:338

```

**Comment by RealRaven2000** [comment]
Alright, guess it is quickfix time again. Please try this version:

[smartTemplate-fx-4.7.1pre25.zip](https://github.com/user-attachments/files/16598414/smartTemplate-fx-4.7.1pre25.zip)

----


To try the version above download zip file and drag the file into Thunderbird Add-ons Manager (do not extract contents, it won't install like that)

**Comment by axxisd** [comment]
This commit works as expected. I also tested to make sure the templates for followups and replies were still in working order, just in case.

Thank you for the quick response and fix!

**Comment by RealRaven2000** [comment]
no worries. I will close the issue once I release it the code the public.

**Comment by RealRaven2000** [comment]
Fixed in 4.8 - published 03/09/2024
